docker-compose/haproxy/haproxy.cfg

```diff
@@ -0,0 +1,62 @@
+global
+log 127.0.0.1   local0
+log 127.0.0.1   local1 notice
+maxconn 4096
+tune.ssl.default-dh-param 2048
+daemon
+
+defaults
+    log         global
+    option      dontlognull
+    mode        http
+
+# HTTP specific timeouts
+    timeout connect     5s
+    timeout client      5s
+    timeout server      5s
+# Websocket specific timeouts
+    timeout tunnel      2h
+
+frontend kuzzle_https
+    bind *:7511 ssl crt /etc/ssl/private/snakeoil.pem
+    default_backend https
+    mode        http
+    option      httpclose
+    option      forwardfor
+    tcp-request inspect-delay 500ms
+    tcp-request content accept if HTTP
```

Comment:
Not really needed for Kuzzle implementation, but the protocol convention "requires" to add a header "X-Forwarded-Proto" when the client's request is in HTTPS:
`   reqadd X-Forwarded-Proto:\ https`